packages/headless/src/modules/user-auth/repository/user.db.interface.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@nestjs/common';
+import { UserEntity } from 'src/entity/user';
+import { CreateUserDto, UserEntityResponse } from '../interface/user.interface';
+
+@Injectable()
+export abstract class IUserAuthDB {
+  abstract save: (body: CreateUserDto) => Promise<UserEntityResponse>;
+  abstract findOne: (
+    query: Record<string, string>,
```

Comment:
can you try mongodb $and:[] query here to check whether your type definition is correct?